### PR TITLE
feature(hooks): a new function to clear all callbacks for a given hook

### LINKED ISSUE #5972: Add ability to remove all handlers for a certain hook
It would be nice to be able to remove all hooks for a certain hook/entity combination.

At this moment we have `elgg_unregister_plugin_hook_handler($hook, $entity_type, $callback)` to unregister a certain callback. An easy solution would be to make $callback optional so we can remove all callback for a certain hook.

Usecase: I need to take over the user_hover menu and there is no need for all the hooks that are registered. Currently i have no other option than manually unset all handlers. What if i could call 
`elgg_unregister_plugin_hook_handler('register', 'menu:user_hover');` without the callback so all hooks are unregistered?

### PR DESCRIPTION
**Pull request by jdalsem**
fixes #5972

### CODE REVIEW

File: engine/classes/Elgg/HooksRegistrationService.php [hunk 1]
```diff
@@ -110,6 +110,25 @@ public function unregisterHandler($name, $type, $callback) {
...
+	 * Clears all handlers for a specific hook
+	 *
+	 * @param string   $name
+	 * @param string   $type
+	 *
+	 * @return void
+	 * @access private
+	 */
+	public function clearHandlers($name, $type) {
+		if (empty($this->registrations[$name][$type])) {
```
Comment (mrclay) [inline engine/classes/Elgg/HooksRegistrationService.php]: We can't just unset this part of the array?

File: engine/lib/elgglib.php [hunk 2]
```diff
@@ -726,6 +726,19 @@ function elgg_unregister_plugin_hook_handler($hook, $entity_type, $callback) {
 }
 
 /**
+ * Clears all callback registrations for a plugin hook.
+ *
+ * @param string $hook        The name of the hook
+ * @param string $entity_type The name of the type of entity (eg "user", "object" etc)
```
Comment (mrclay) [inline engine/lib/elgglib.php]: It's just $type and "Hook type"

File: engine/classes/Elgg/HooksRegistrationService.php (hunk 1 above)
Comment (jdalsem) [inline engine/classes/Elgg/HooksRegistrationService.php]: i think so...

**Comment by jdalsem** [comment]
took care of the comments

**Comment by mrclay** [comment]
LGTM. This could go in 1.12 but would require changes and merge difficulty. Up to you.

**Pull request merged**